app/src/main/java/org/simple/clinic/editpatient/PatientEditScreenController.kt

```diff
@@ -378,4 +384,110 @@ class PatientEditScreenController @Inject constructor(
           }
         }
   }
+
+  private fun closeScreenWithoutSaving(events: Observable<UiEvent>): Observable<UiChange> {
+    val ongoingEntryChanges = events
+        .ofType<OngoingEditPatientEntryChanged>()
+        .map { it.ongoingEditPatientEntry }
+
+    val patientUuidStream = events.ofType<PatientEditScreenCreated>()
+        .map { it.patientUuid }
+
+    val savedPatient = patientUuidStream
+        .flatMap { patientRepository.patient(it).take(1).unwrapJust() }
+        .replay()
+        .refCount()
+
+    val savedPatientAddress = savedPatient
+        .flatMap { patientRepository.address(it.addressUuid).take(1).unwrapJust() }
+        .replay()
+        .refCount()
+
+    val savedNumbers = patientUuidStream
+        .flatMap { patientRepository.phoneNumbers(it).take(1) }
+        .replay()
+        .refCount()
+
```

Comment:
It might; I'll revisit it in a future PR. I might try extracting the `PatientProfile` stream separately and provide it to the functions which need it via a parameter.

Reply:
👍